pkgs/halo2-eth-membership/src/eth_membership.rs

```diff
@@ -675,6 +648,107 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_eff_ecdsa_verification() -> Result<()> {
+        let path = "configs/eth_membership.config";
+        let circuit_params: CircuitConfig = serde_json::from_reader(
+            File::open(path)
+                .map_err(|e| anyhow!(e))
+                .with_context(|| format!("The circuit config file does not exist: {}", path))?,
+        )
+        .map_err(|e| anyhow!(e))
+        .with_context(|| format!("Failed to read the circuit config file: {}", path))?;
+
+        let (ecdsa_inputs, test_inputs) = mock_eff_ecdsa_input(PRIV_KEY)?;
+
+        let merkle_proof = mock_merkle_proof(&test_inputs.address)?;
+
+        let eth_membership_inputs = EthMembershipInputs::<
+            secp256k1::Fp,
+            secp256k1::Fq,
+            Secp256k1Affine,
+        >::new(ecdsa_inputs, merkle_proof);
+
+        let mut halo2_wasm = Halo2Wasm::new();
+
+        halo2_wasm.config(circuit_params);
+
+        let mut circuit =
+            EthMembershipCircuit::<secp256k1::Fp, secp256k1::Fq, Secp256k1Affine>::new(
+                &halo2_wasm,
+                eth_membership_inputs,
+            )?;
+
+        circuit.verify_membership();
+
+        let params = ParamsKZG::<E>::setup(K, OsRng);
+
+        // Load params
+        halo2_wasm.load_params(&serialize_params_to_bytes(&params));
+
+        // Generate VK
+        halo2_wasm.gen_vk();
+
+        // Generate PK
+        halo2_wasm.gen_pk();
+
+        // Get the public instance inputs
+        let instances = halo2_wasm.get_instance_values_ext(INSTANCE_COL)?;
+
+        let start = Instant::now();
+
+        // Verify Eff ECDSA
+        println!("Verifying Eff ECDSA Proof");
+
+        let is_eff_ecdsa_valid = verify_efficient_ecdsa(
+            test_inputs.msg_hash,
+            test_inputs.r,
+            test_inputs.is_y_odd,
+            &instances,
+        )?;
+
+        println!("- Is Eff ECDSA valid? {}", is_eff_ecdsa_valid);
```

Comment:
```suggestion
```
I don't think we need this as we have already the assertion

Reply:
That is just meant to see on the terminal after the test.